service-base/src/main/java/org/eclipse/hono/service/credentials/CompleteCredentialsService.java

```diff
@@ -105,37 +108,40 @@ default void getAll(final String tenantId, final String deviceId, final Span spa
      *         format defined by the Credentials API.</li>
      *         <li><em>404 Not Found</em> if no credentials of the given type and auth-id exist.</li>
      *         </ul>
-     * @throws NullPointerException if any of the parameters is {@code null}.
+     * @throws NullPointerException if any of the parameters except resourceVersion is {@code null}.
      * @see <a href="https://www.eclipse.org/hono/api/credentials-api/#update-credentials">
      *      Credentials API - Update Credentials</a>
      */
-    void update(String tenantId, JsonObject credentialsObject, Handler<AsyncResult<CredentialsResult<JsonObject>>> resultHandler);
+    void update(String tenantId, JsonObject credentialsObject, Optional<String> resourceVersion, Handler<AsyncResult<CredentialsResult<JsonObject>>> resultHandler);
 
     /**
      * Removes credentials by authentication identifier and type.
      *
      * @param tenantId The tenant the device belongs to.
      * @param type The type of credentials to remove.
      * @param authId The authentication identifier to remove credentials for.
+     * @param resourceVersion The identifier of the resource version to update.
      * @param resultHandler The handler to invoke with the result of the operation.
+     * @deprecated removed in hono 1.0 API.
      *         <p>
      *         The <em>status</em> will be
      *         <ul>
      *         <li><em>204 No Content</em> if the credentials matching the criteria have been removed.</li>
      *         <li><em>404 Not Found</em> if no credentials match the given criteria.</li>
      *         </ul>
-     * @throws NullPointerException if any of the parameters is {@code null}.
+     * @throws NullPointerException if any of the parameters except resourceVersion is {@code null}.
```

Comment:
I think this is not completely correct. If the `resourceVersion` is `null` I would expect a NPE. However, if it is "empty" it should be "ok".

Reply:
Agreed. Since I changed to using Optional having a null value will raise NPE. It's updated.